posts/2024-02-27-24.0.0.2.adoc

```diff
@@ -0,0 +1,332 @@
+For `mpReactiveMessaging-3.0`, the default value of the option is `true`,
+
+==== Manage asynchronous tasks with context.service
+
+The `context.service` attribute specifies the context service that is used for asynchronous tasks.
+The value of the `context.service` attribute is a reference to the `id` attribute of a `context.service` instance that is defined in `server.xml` file.
+
+In the following `server.xml` file, three different context services are defined with unique IDs.
+
+[source,xml]
+----
+<contextService id=“rst”/>
+<contextService id=“uvw”/>
+<contextService id=“xyz”/>
+----
+
+In the application's `microprofile-config.properties` file, the first context service is set on the connector. The application has three channels. The `def` channel does not specify its own `context.service` instance, so it uses the one that is defined on the connector. The second and third channels define and use their own services.
+
+[source,properties]
+----
+mp.messaging.connector.liberty-kafka.context.service=rst
+
+mp.messaging.incoming.def.connector=liberty-kafka
+mp.messaging.incoming.foo.connector=liberty-kafka
+mp.messaging.incoming.foo.context.service=uvw
+mp.messaging.outgoing.bar.connector=liberty-kafka
+mp.messaging.outgoing.bar.context.service=xyz
+----
+
+If a version of the link:{url-prefix}/docs/latest/reference/feature/concurrent-3.0.html[Jakarta Concurrency] feature is enabled in the `server.xml` file, the default context service is used. If this feature is not enabled, the built-in Open Liberty context service is used with a set list of context types to capture and apply around asynchronous tasks.
+
+For more information, see the following resources:
+
+- link:{url-prefix}/docs/latest/liberty-kafka-connector.html[Optimizing Kafka integration with MicroProfile Messaging]
+- link:https://download.eclipse.org/microprofile/microprofile-reactive-messaging-3.0/microprofile-reactive-messaging-spec-3.0.html[Reactive Messaging specification]
+- link:https://download.eclipse.org/microprofile/microprofile-reactive-streams-operators-3.0/microprofile-reactive-streams-operators-spec-3.0.html[Reactive Streams Operators specification]
+
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+[#tescont]
+== New guide: "Building true-to-production integration tests with Testcontainers"
+
+A new guide, link:https://openliberty.io/guides/testcontainers.html[Building true-to-production integration tests with Testcontainers], is published under the link:https://openliberty.io/guides/#test[Test] category.
+In this guide, you’ll learn how to write true-to-production integration tests for Java microservices by using link:https://www.testcontainers.org/[Testcontainers] and JUnit. You can also run this guide in a hosted environment, without having to worry about local setup and prerequisites. To access the cloud-hosted version of this guide, click the **Run in cloud** button in the guide code pane.
+
+
+[#CVEs]
+== Security vulnerability (CVE) fixes in this release
```

Comment:
It looks like before 24.0.0.1 the "For a list of past security vulnerability fixes..." statement wasn't put in the blog at all unless there were CVEs fixed in that particular release. And in that case the statement was just put at the end of the "Security vulnerability (CVE) fixes in this release" section.

I'd lean towards not having the CVE section at all if there are no CVEs fixed in the release. Having a link to this section at the top of the page makes me think there were CVEs that we fixed. It's not until I click that link or scroll down here that I find out there are none. For these cases where there are no CVE fixes, I'd probably stick with what was done in 24.0.0.1 and have the "For a list of past security vulnerability fixes..." statement after the "View the list of fixed bugs..." statement at the top of the page.

Reply:
fixed in #3597